arch/cortex-v7m/README.md

```diff
@@ -0,0 +1,10 @@
+Cortex-M v7m Architecture
+=========================
```

Comment:
Thoughts on adding the `cortex-v7m` crate (copied over from Slack):

Upsides:
- It removes a nightly feature
- It's consequential, given that we have a common cortex-m crate and a cortex-m0 crate, which contains v6m assembly already
- (I'm biased, but) I think the cortexm::CortexMVariant  encapsulates these arch differences very nicely across crates, without involving guesswork as to what the linker's gonna stick together

Downsides:
- Yet another crate
- Increases compilation time slightly

I'm in favor, esp. given we want to compile on stable.